test/t/repeated_packed_double/reader_test_cases.cpp

```diff
@@ -47,6 +47,26 @@ TEST_CASE("read repeated packed double field") {
             REQUIRE(*it++ == std::numeric_limits<double>::min());
             REQUIRE(*it++ == std::numeric_limits<double>::max());
             REQUIRE(it == it_range.end());
+
+            it = it_range.begin();
+            auto it2 = it + 1;
+            REQUIRE(it2 > it);
+            REQUIRE(it < it2);
+            REQUIRE(it <= it_range.begin());
+            REQUIRE(it >= it_range.begin());
+            REQUIRE(*it2 == 0.0);
+            auto it3 = 1 + it;
+            REQUIRE(*it3 == 0.0);
+            auto it4 = it2 - 1;
+            REQUIRE(*it4 == 17.34);
```

Comment:
Floating point comparisons should use `Approx()` https://github.com/catchorg/Catch2/blob/master/docs/assertions.md#floating-point-comparisons . Aargh. I am seeing I don't have that in the original code either!